Dockerfile.ci

```diff
@@ -315,6 +315,9 @@ WORKDIR ${AIRFLOW_SOURCES}
 
 ENV PATH="${HOME}:${PATH}"
 
+# Needed to stop Gunicorn from crashing when /tmp is now mounted from host
+ENV GUNICORN_CMD_ARGS="--worker-tmp-dir /opt/airflow/tmp"
```

Comment:
Oh wait.. but that means it will overwrite existing cmd args?

Reply:
Because we are configuring a lot of that via `[webserver]` section and some of it also via breeze args I think.

Reply:
good question !

Reply:
I used the env var approach because that is the easiest way if you do not have access to the env directly.

If we would just pass `--worker-tmp-dir` as a breeze cli argument that would probably be better to handle?

Reply:
Can I ask a different question: why do we need to mount /tmp from the host?

Reply:
Other option. Set this in the entrypoint as:


```
export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-} --worker-tmp-dir /opt/airflow/tmp"

Reply:
It's ok. But I change it to /dev/shm -> just tested that it works nicely

Reply:
@ashb - the root cause was the removal of Java from the CI image. 

@mik-laj spend quite some time after it was removed to make it works and that involved mapping /tmp from host to both Breeze image and to Java run via the docker image and forwarded docker socket.

This is all because we have automated system tests in Google Provder. Previously we just run java in Breeze when we needed it  and we used java to prepare packages to be sent to DataFlow (it needs to package java code and some jars and submit them to DataFlow) 

Previously when java was embedded in the image it "just worked". But when you removed it from the image (which was a good thing BTW.) we needed to find another solution. We used the approach that I piloted with the gcloud CLI - where we have script that runs external tool via Docker run in the docker engine via the forwarded socket. The problem with that was that we had to pass some of the temporary files we created in Python unit tests to the java command line (that was also similar problem for other code ours where we uploaded temporary files to GCS via gcloud command line). 

The easiest way was to map ${AIRFLOW_SOURCES}/tmp to both - the CI airflow container  and to all the tools we run via the forwarded socket. This way the /tmo (we also did it for  /opt/airflow) is shared between the CI container and the gcloud, java, terraform .and all the other tools which we now do not have to embed into the CI image - they are all pulled on demand via docker.

Since we switched to run those tools "docker via

Reply:
> I used the env var approach because that is the easiest way if you do not have access to the env directly.

@feluelle @ashb -> the ENV approach is good to go. Seems this has the right characteristics.  - it will take precedence over other methods (conf/commandline) but you can still specify other options. So I think for CI image it's good to go -  but for production image definitely this approach is better:

```
export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-} --worker-tmp-dir /dev/shm"
```

I also decided to use /dev/shm as suggested by https://docs.gunicorn.org/en/stable/faq.html#how-do-i-avoid-gunicorn-excessively-blocking-in-os-fchmod: